test/integration/entry-integration.js

```diff
@@ -419,4 +419,131 @@ describe('Entry Api', () => {
+            type: 'ResourceLink',
+            validations: [],
+          },
+          allowedResources: [
+            {
+              type: 'Contentful:Entry',
+              source: 'crn:contentful:::content:spaces/6mqcevu5a50r',
+              contentTypes: ['testCtxSpace'],
+            },
+          ],
+        },
+      ],
+    }
+
+    let entryData = {
+      fields: {
+        title: { 'en-US': 'this is the title' },
+        multiRefXSpace: {
+          'en-US': [
+            {
+              sys: {
+                type: 'ResourceLink',
+                linkType: 'Contentful:Entry',
+                urn: 'crn:contentful:::content:spaces/6mqcevu5a50r/entries/4zimQzVMxDsSX607PCfo2u',
+              },
+            },
+          ],
+        },
+      },
+    }
+
+    before(async () => {
+      // default space has xspace enabled in it through the feature flags
+      xSpaceEnabledSpace = await getDefaultSpace()
+      xSpaceEnabledEnvironment = await xSpaceEnabledSpace.getEnvironment('master')
+      xSpaceDisabledSpace = await createTestSpace(initClient(), 'Entry')
+      xSpaceDisabledEnvironment = await createTestEnvironment(
+        xSpaceDisabledSpace,
+        'Testing Environment'
+      )
+      contentTypeDisabledSpace = await xSpaceDisabledEnvironment.createContentTypeWithId(
+        generateRandomId('test-content-type'),
+        contentTypeData
+      )
+      await contentTypeDisabledSpace.publish()
+
+      contentTypeEnabledSpace = await xSpaceEnabledEnvironment.createContentTypeWithId(
+        generateRandomId('test-content-type'),
+        contentTypeData
+      )
+      await contentTypeEnabledSpace.publish()
+    })
+
+    after(async () => {
+      await contentTypeDisabledSpace
+        .unpublish()
+        .then((unpublishedContentType) => unpublishedContentType.delete())
+      await contentTypeEnabledSpace
+        .unpublish()
+        .then((unpublishedContentType) => unpublishedContentType.delete())
+      await xSpaceDisabledSpace.delete()
+    })
+
+    test('Blocks publishing an entry with xspace references if the feature is disabled for the space', async () => {
+      return xSpaceDisabledEnvironment
+        .createEntry(contentTypeDisabledSpace.sys.id, entryData)
+        .then((entry) => {
+          expect(entry.isDraft(), 'entry is in draft').ok
+          expect(entry.fields.title['en-US']).equals('this is the title', 'original title')
+          return entry.publish().catch((accessDeniedError) => {
+            let errorMessage = JSON.parse(accessDeniedError.message)
+            expect(accessDeniedError.name).equals('AccessDenied', 'Access Denied Error')
+            expect(errorMessage.status).equals(403, '403 forbidden status')
+            expect(errorMessage.details.reasons).equals(
+              'Cross space links feature is not enabled for this space',
+              'reason explained'
+            )
+            return xSpaceDisabledEnvironment.deleteEntry(entry.sys.id)
```

Comment:
I think this clean up doesn't belong here. You're not asserting deletion. And you're already cleaning up in the after block.

Reply:
I was trying to keep the same test structures as the previous ones and deleting the entry here is easier because we already have the entry at hand

Reply:
but I guess I can do it in the after block so all the clean up would be in the same place